plugin/inlay_hint.py

```diff
@@ -0,0 +1,127 @@
+        if not session:
+            return
+        text_edits = inlay_hint.get('textEdits')
+        if not text_edits:
+            return
+        for sv in session.session_views_async():
+            sv.remove_inlay_hint_phantom(phantom_uuid)
+        apply_text_edits_to_view(text_edits, self.view)
+
+
+def get_inlay_hint_html(view: sublime.View, inlay_hint: InlayHint, session: Session, phantom_uuid: str) -> str:
+    tooltip = format_inlay_hint_tooltip(view, inlay_hint.get("tooltip"))
+    label = format_inlay_hint_label(view, inlay_hint["label"])
+    margin_left = "0.6rem" if inlay_hint.get("paddingLeft", False) else "0"
+    margin_right = "0.6rem" if inlay_hint.get("paddingRight", False) else "0"
+    font = view.settings().get('font_face') or "monospace"
+    html = """
+    <body id="lsp-inlay-hint">
+        <style>
+            .inlay-hint {{
+                color: color(var(--foreground) alpha(0.6));
+                background-color: color(var(--foreground) alpha(0.08));
+                border-radius: 4px;
+                margin-left: {margin_left};
+                margin-right: {margin_right};
+                padding: 0.05em 4px;
+                font-size: 0.9em;
+                font-family: {font};
+            }}
+
+            .inlay-hint a {{
+                color: color(var(--foreground) alpha(0.6));
+                text-decoration: none;
+            }}
+        </style>
+        <div class="inlay-hint" title="{tooltip}">
+    """.format(
+        margin_left=margin_left,
+        margin_right=margin_right,
+        tooltip=tooltip,
+        font=font
+    )
+    can_resolve_inlay_hint = session.has_capability('inlayHintProvider.resolveProvider')
+    apply_text_edits_command = sublime.command_url('lsp_inlay_hint_apply_text_edits', {
+        'session_name': session.config.name,
+        'inlay_hint': inlay_hint,
+        'phantom_uuid': phantom_uuid
+    })
+    can_apply_text_edits = inlay_hint.get('textEdits') or (not inlay_hint.get('textEdits') and can_resolve_inlay_hint)
+    if can_apply_text_edits:
+        html += '<a href="{apply_text_edits_command}">'.format(apply_text_edits_command=apply_text_edits_command)
+    html += label
+    if can_apply_text_edits:
+        html += '</a>'
+    html += """
+        </div>
+    </body>
+    """
+    return html
+
+
+def format_inlay_hint_tooltip(view: sublime.View, tooltip: Optional[Union[str, MarkupContent]]) -> str:
+    if isinstance(tooltip, str):
+        return html.escape(tooltip)
+    elif isinstance(tooltip, dict):  # MarkupContent
+        return minihtml(view, tooltip, allowed_formats=FORMAT_MARKUP_CONTENT)
+    else:
+        return ""
+
+
+def format_inlay_hint_label(view: sublime.View, label: Union[str, List[InlayHintLabelPart]]) -> str:
+    if isinstance(label, str):
+        return html.escape(label)
+    else:
+        return "".join("<div title=\"{tooltip}\">{value}</div>".format(
```

Comment:
In combination with `can_apply_text_edits==True` this will not work because there can be no block elements inside an inline element. I saw that this causes hint content to be hidden.

Reply:
Noticed this as well, in [a9e268b](https://github.com/sublimelsp/LSP/pull/2018/commits/a9e268be84a7436f16245ba817656194ea5f509d) I made sure that this does not happen